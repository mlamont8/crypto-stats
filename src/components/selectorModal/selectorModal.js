import React from 'react';
import { connect } from "react-redux";


class SelectorModal extends React.Component {

    render() {
        const { market, convertFrom, convertTo, marketArray, fromArray, toArray } = this.props;
        let currentArray = {};
        console.log({ market });
        console.log({ marketArray });
        if (market === false) { currentArray = marketArray; }
        console.log('currentArray', currentArray);
        return (
            <div className="modalContainer">
                <div className="modalContent">
                    Modal Content
                <button type="button" onClick={this.props.handleClose}>Close</button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        market: state.searchTerm.market,
        convertFrom: state.searchTerm.convertFrom,
        convertTo: state.searchTerm.convertTo,
        marketArray: state.searchArrays.marketArray,
        fromArray: state.searchArrays.convertFrom,
        toArray: state.searchArrays.convertTo,
    };
};

// const mapDispatchToProps = dispatch => {
//     return {
//         changeNotification: () => {
//             dispatch(notificationOption());
//         }
//     };
// };

export default connect(mapStateToProps, null)(SelectorModal);