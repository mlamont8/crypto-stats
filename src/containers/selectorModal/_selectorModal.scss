.modalContainer {
    position: fixed;
    justify-content: center;
    align-items: center;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.6);
    z-index: 3;
}


.modalContent {
    position: fixed;
    background: $grayBlue;
    width: 80%;
    max-height: 80vh;
    border-radius: 10px;
    overflow-y: auto;
    top:50%;
    left:50%;
    transform: translate(-50%,-50%);
    color: black;
}

  .modalHeader{
    background-color: $grayBlue;
    color: #dddddd;
    height: 50px;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
    text-decoration: underline;
    div {
      flex: 1;
      font-size: 2em;
    }
    
  }

  .modalFooter{
    background-color: $grayBlue;
    color: $white;
    height: 50px;
  }

  .modalInstructions{
    height: 50px;
    font-size: 1.8em;
    background-color: $grayBlue;
    color: #dddddd;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 50px 10px;
    div {
      text-align: center;
    }
  }
    
    button{
      color: $grayBlue;
      font-size: .8em;
      margin: 0 10px;
      border-radius: 5px;
      transition: all .2s ease;
    }
    button:hover {
      background-color: $grayBlue; 
      color: #dddddd;
    }


  .coinError {
    color: red;
  }

  .selectorItems {
    
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-around;
    
    div{
      color: black;
      list-style: none;
      flex: 0 1 180px;
      text-align: center;
      font-size: 20px;
      margin-bottom: 8px;
    }
      a {
       color: $selectText;
     }
  }
  
// Animated ellipsis
  .searching:after{
    content: "\2026";
    overflow: hidden;
    display: inline-block;
    vertical-align: bottom;
     animation:
      blinking steps(4, end) 900ms infinite;
    width: 0px;
    }

  @keyframes blinking {
   
    to {width: 1.25em}
  }
  
