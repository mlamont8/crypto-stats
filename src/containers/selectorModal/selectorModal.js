import React from 'react';
import { connect } from "react-redux";
import { SelectData, modalState, idUpdate } from "../../actions";

// Receives props from state tree

export class SelectorModal extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            id: '',
        }
        this.onOptionClick = this.onOptionClick.bind(this);
    }

    // Updates to new id after click
    onOptionClick(item) {

        const id = this.props.id;
        // send item with current id
        this.props.optionClick(id, item)
        if (id === 'market') {

            // set id to convertfrom
            this.props.idUpdate('convertFrom')

        } else if (id === 'convertFrom') {
            // set id to convertTo
            this.props.idUpdate('convertTo')
        }
        else {
            // reset id
            this.props.idUpdate('market')
            // Set modal to closed
            this.props.modalStatus(false)
        }
    }

    render() {
        // get object from state store
        const { currentArray } = this.props;
        return (
            <div className="modalContainer">
                <div className="modalContent">
                    <div className="selectorItems">
                        {currentArray.sort().map(item =>
                            <div key={item}>
                                <a onClick={this.onOptionClick.bind(this, { item }.item)}>{item}</a>
                            </div>)}
                    </div>
                    <button type="button" onClick={this.props.modalStatus.bind(this, false)}>Close</button>
                    <button type="button" onClick={this.props.searchReset}>Reset</button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        currentArray: state.searchArrays.currentArray,
        id: state.searchArrays.currentID
    };
};

const mapDispatchToProps = dispatch => {
    return {
        optionClick: (id, selection) => {
            dispatch(SelectData(id, selection));
        },
        search: () => {
            dispatch({ type: "SEARCH_REQUEST" });
        },
        modalStatus: (toggle) => {
            dispatch(modalState(toggle));
        },
        searchReset: () => {
            dispatch({ type: "NEW_RESET" });
        },
        idUpdate: (id) => {
            dispatch(idUpdate(id));
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(SelectorModal);