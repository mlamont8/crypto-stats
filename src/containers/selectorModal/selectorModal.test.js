import React from "react";
import { shallow } from "enzyme";
import { SelectorModal } from "./selectorModal";

describe("Selector Modal", () => {
    const currentArray = [0, 1];
    const component = shallow(<SelectorModal currentArray={currentArray} closeModal={jest.fn} />)

    it("renders without crashing", () => {
        expect(component).toMatchSnapshot();
    });
})