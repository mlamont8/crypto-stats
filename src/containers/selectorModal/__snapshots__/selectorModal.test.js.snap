// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector Modal renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectorModal
    closeModal={[Function]}
    currentArray={
        Array [
            0,
            1,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modalContent"
>
        <div
                className="modalInstructions"
        >
                <div>
                        Choose your conversion currency
                </div>
                <div>
                        <button
                                onClick={undefined}
                                type="button"
                        >
                                Reset
                        </button>
                        <button
                                onClick={[Function]}
                                type="button"
                        >
                                Close
                        </button>
                </div>
        </div>
        <div
                className="modalHeader"
        >
                <div>
                         
                </div>
                <div />
        </div>
        <div
                className="modalAlerts"
        >
                Alerts
        </div>
        <div
                className="selectorItems"
        >
                <div>
                        <a
                                onClick={[Function]}
                        />
                </div>
                <div>
                        <a
                                onClick={[Function]}
                        >
                                1
                        </a>
                </div>
        </div>
        <div
                className="modalFooter"
        >
                <button
                        onClick={[Function]}
                        type="button"
                >
                        Close
                </button>
                <button
                        onClick={undefined}
                        type="button"
                >
                        Reset
                </button>
        </div>
</div>,
      "className": "modalContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modalInstructions"
>
            <div>
                        Choose your conversion currency
            </div>
            <div>
                        <button
                                    onClick={undefined}
                                    type="button"
                        >
                                    Reset
                        </button>
                        <button
                                    onClick={[Function]}
                                    type="button"
                        >
                                    Close
                        </button>
            </div>
</div>,
          <div
            className="modalHeader"
>
            <div>
                         
            </div>
            <div />
</div>,
          <div
            className="modalAlerts"
>
            Alerts
</div>,
          <div
            className="selectorItems"
>
            <div>
                        <a
                                    onClick={[Function]}
                        />
            </div>
            <div>
                        <a
                                    onClick={[Function]}
                        >
                                    1
                        </a>
            </div>
</div>,
          <div
            className="modalFooter"
>
            <button
                        onClick={[Function]}
                        type="button"
            >
                        Close
            </button>
            <button
                        onClick={undefined}
                        type="button"
            >
                        Reset
            </button>
</div>,
        ],
        "className": "modalContent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Choose your conversion currency
</div>,
              <div>
                <button
                                onClick={undefined}
                                type="button"
                >
                                Reset
                </button>
                <button
                                onClick={[Function]}
                                type="button"
                >
                                Close
                </button>
</div>,
            ],
            "className": "modalInstructions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Choose your conversion currency",
              },
              "ref": null,
              "rendered": "Choose your conversion currency",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    onClick={undefined}
                    type="button"
>
                    Reset
</button>,
                  <button
                    onClick={[Function]}
                    type="button"
>
                    Close
</button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Reset",
                    "onClick": undefined,
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Reset",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Close",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Close",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                 
</div>,
              <div />,
            ],
            "className": "modalHeader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Alerts",
            "className": "modalAlerts",
          },
          "ref": null,
          "rendered": "Alerts",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <a
                                onClick={[Function]}
                />
</div>,
              <div>
                <a
                                onClick={[Function]}
                >
                                1
                </a>
</div>,
            ],
            "className": "selectorItems",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  onClick={[Function]}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": 0,
                "type": "a",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  onClick={[Function]}
>
                  1
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": 1,
                "type": "a",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
                type="button"
>
                Close
</button>,
              <button
                onClick={undefined}
                type="button"
>
                Reset
</button>,
            ],
            "className": "modalFooter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Close",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Close",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset",
                "onClick": undefined,
                "type": "button",
              },
              "ref": null,
              "rendered": "Reset",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modalContent"
>
          <div
                    className="modalInstructions"
          >
                    <div>
                              Choose your conversion currency
                    </div>
                    <div>
                              <button
                                        onClick={undefined}
                                        type="button"
                              >
                                        Reset
                              </button>
                              <button
                                        onClick={[Function]}
                                        type="button"
                              >
                                        Close
                              </button>
                    </div>
          </div>
          <div
                    className="modalHeader"
          >
                    <div>
                               
                    </div>
                    <div />
          </div>
          <div
                    className="modalAlerts"
          >
                    Alerts
          </div>
          <div
                    className="selectorItems"
          >
                    <div>
                              <a
                                        onClick={[Function]}
                              />
                    </div>
                    <div>
                              <a
                                        onClick={[Function]}
                              >
                                        1
                              </a>
                    </div>
          </div>
          <div
                    className="modalFooter"
          >
                    <button
                              onClick={[Function]}
                              type="button"
                    >
                              Close
                    </button>
                    <button
                              onClick={undefined}
                              type="button"
                    >
                              Reset
                    </button>
          </div>
</div>,
        "className": "modalContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="modalInstructions"
>
              <div>
                            Choose your conversion currency
              </div>
              <div>
                            <button
                                          onClick={undefined}
                                          type="button"
                            >
                                          Reset
                            </button>
                            <button
                                          onClick={[Function]}
                                          type="button"
                            >
                                          Close
                            </button>
              </div>
</div>,
            <div
              className="modalHeader"
>
              <div>
                             
              </div>
              <div />
</div>,
            <div
              className="modalAlerts"
>
              Alerts
</div>,
            <div
              className="selectorItems"
>
              <div>
                            <a
                                          onClick={[Function]}
                            />
              </div>
              <div>
                            <a
                                          onClick={[Function]}
                            >
                                          1
                            </a>
              </div>
</div>,
            <div
              className="modalFooter"
>
              <button
                            onClick={[Function]}
                            type="button"
              >
                            Close
              </button>
              <button
                            onClick={undefined}
                            type="button"
              >
                            Reset
              </button>
</div>,
          ],
          "className": "modalContent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  Choose your conversion currency
</div>,
                <div>
                  <button
                                    onClick={undefined}
                                    type="button"
                  >
                                    Reset
                  </button>
                  <button
                                    onClick={[Function]}
                                    type="button"
                  >
                                    Close
                  </button>
</div>,
              ],
              "className": "modalInstructions",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Choose your conversion currency",
                },
                "ref": null,
                "rendered": "Choose your conversion currency",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      onClick={undefined}
                      type="button"
>
                      Reset
</button>,
                    <button
                      onClick={[Function]}
                      type="button"
>
                      Close
</button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Reset",
                      "onClick": undefined,
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Reset",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Close",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "Close",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                   
</div>,
                <div />,
              ],
              "className": "modalHeader",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Alerts",
              "className": "modalAlerts",
            },
            "ref": null,
            "rendered": "Alerts",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <a
                                    onClick={[Function]}
                  />
</div>,
                <div>
                  <a
                                    onClick={[Function]}
                  >
                                    1
                  </a>
</div>,
              ],
              "className": "selectorItems",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    onClick={[Function]}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "a",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    onClick={[Function]}
>
                    1
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "a",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  onClick={[Function]}
                  type="button"
>
                  Close
</button>,
                <button
                  onClick={undefined}
                  type="button"
>
                  Reset
</button>,
              ],
              "className": "modalFooter",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Close",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Close",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Reset",
                  "onClick": undefined,
                  "type": "button",
                },
                "ref": null,
                "rendered": "Reset",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
